src/mme-app/utils/mmeNasUtils.cpp

```diff
@@ -569,10 +570,52 @@ int MmeNasUtils::parse_nas_pdu(s1_incoming_msg_data_t* msg_data, unsigned char *
         }
 
         case NAS_TAU_REQUEST:
-		{
+	{
             log_msg(LOG_INFO, "NAS_TAU_REQUEST recvd\n");
-            break;
-		}
+            nas->elements_len = 2;
+            nas->elements = (nas_pdu_elements *)calloc(sizeof(nas_pdu_elements), 2);
+
+            uint8_t elem_id = msg[0];
+            uint8_t index = 0;
+            uint8_t datalen = 0;
+
+            while(msg != msg_end)
+            {
+                //Handling only UE Network Capability and UE Additional Sec Capab as a part of NSA Support
+                elem_id = msg[0];
+                switch(elem_id)
+                {
+                    case NAS_IE_TYPE_UE_NETWORK_CAPABILITY:
+                    {
+                        log_msg(LOG_DEBUG, "NAS TAU REQ - UE Network capability : Handling.\n");
+                        nas->elements[index].msgType = NAS_IE_TYPE_UE_NETWORK_CAPABILITY;
+                        msg++; //skipping Element ID
+                        datalen = msg[0];
+                        nas->elements[index].pduElement.ue_network.len = datalen;
+                        log_msg(LOG_INFO, "UE NET CAP Length %d", datalen);
+                        msg++; 
+                        memcpy(&(nas->elements[index].pduElement.ue_network.u.octets), msg, datalen);
+                        msg += datalen;
+                        index++;
+                        break;
+                    }
+                    case NAS_IE_TYPE_UE_ADDITIONAL_SECURITY_CAPABILITY:
+                    {
+                        log_msg(LOG_DEBUG, "NAS TAU REQ - UE Additional Security capability : Handling.\n");
+                        nas->elements[index].msgType = NAS_IE_TYPE_UE_ADDITIONAL_SECURITY_CAPABILITY;
+                        msg++; //skipping Element ID
+                        datalen = msg[0];
+                        msg++;
+                        memcpy(&(nas->elements[index].pduElement.ue_add_sec_capab), msg, datalen);
+                        msg += datalen;
+                        index++;
+                        break;
+                    }
+                    default:
+                        msg++;
```

Comment:
This does not seem to be correct. Should we not skip the whole AVP/IE else Mme will crash. Thes

Reply:
Taken care in d588d53